lld/COFF/Driver.cpp

```diff
@@ -2601,7 +2601,10 @@ void LinkerDriver::linkerMain(ArrayRef<const char *> argsArr) {
     createECExportThunks();
 
   // Resolve remaining undefined symbols and warn about imported locals.
-  ctx.symtab.resolveRemainingUndefines();
+  if (ctx.symtab.resolveRemainingUndefines()) {
+    run();
+    ctx.symtab.resolveRemainingUndefines();
```

Comment:
I'm pondering whether it would make sense to convert it into a loop,
```
while (ctx.symtab.resolveRemainingUndefines())
  run();
```
That has some charm of genericity to it - but also looks more prone to potentially hang (even though that really shouldn't happen). I don't have a very strong opinion on it either way though.

Reply:
I was pondering about that while writing the code. I guess there isn't really a risk that we end up having more forceLazy happening indefinitely.